kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -25,231 +27,457 @@
 namespace Kratos
 {
 
-    FindIntersectedGeometricalObjectsProcess::FindIntersectedGeometricalObjectsProcess(ModelPart& rPart1, ModelPart& rPart2)
-        : mrModelPart1(rPart1), mrModelPart2(rPart2)
-    {
-    }
+template<class TEntity>
+FindIntersectedGeometricalObjectsProcess<TEntity>::FindIntersectedGeometricalObjectsProcess(
+    ModelPart& rPart1,
+    ModelPart& rPart2
+    ) : mrModelPart1(rPart1),
+        mrModelPart2(rPart2)
+{
+    const Parameters default_parameters = GetDefaultParameters();
+    mThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+FindIntersectedGeometricalObjectsProcess<TEntity>::FindIntersectedGeometricalObjectsProcess(
+    Model& rModel,
+    Parameters ThisParameters
+    ) : mrModelPart1(rModel.GetModelPart(ThisParameters["first_model_part_name"].GetString())),
+        mrModelPart2(rModel.GetModelPart(ThisParameters["second_model_part_name"].GetString())),
+        mThisParameters(ThisParameters)
```

Comment:
Why do we need to save the Parameters as a member variable?

Reply:
For the sake of generality (it makes more sense in #4263)

Reply:
I wouldn't save them if there is no necessity.

Reply:
This is removed, look the outdated label